rust/libnewsboat/src/utils.rs

```diff
@@ -688,6 +691,37 @@ pub fn getcwd() -> Result<PathBuf, io::Error> {
     env::current_dir()
 }
 
+/// Get the lines of text contained in a file.
+pub fn read_text_file(filename: &Path) -> Result<Vec<String>, String> {
+    use std::fs::File;
+    use std::io::BufRead;
+    let file = match File::open(filename) {
+        Ok(f) => f,
+        Err(reason) => {
+            return Err(fmt!(
+                &gettext("Failed to open file (%s)"),
+                reason.to_string()
+            ))
+        }
+    };
```

Comment:
I think this could be replaced by `map_err`:

```rust
let file = File::open(filename)
    .map_err(|reason| fmt!(&gettext("Failed to open file (%s)"), reason.to_string()))?;
```

Reply:
Nice improvement, thank you for the suggestion.
Applied in https://github.com/newsboat/newsboat/pull/1332/commits/bba28abe66187ce9504892440c7ec49b8b82ed6d.